## Task 8 — Revenue per Customer by Plan

### Business Question

What is the average monthly revenue per customer for each subscription plan?

### Analysis

Group customers by subscription plan, calculate the total monthly revenue and number of customers for each plan, then calculate the average monthly revenue per customer and sort the results in descending order.

In [1]:
import pandas as pd
import numpy as np
customers = pd.read_csv("starlink_customers.csv")
subscriptions = pd.read_csv("subscriptions.csv")
starlink = customers.merge(
    subscriptions,
    left_on = "unique_id",
    right_on = "customer_id",
    how = "left"
)
total_revenue_by_plan = starlink.groupby("plan_name").agg(
    total_revenue = ("monthly_fee","sum"),
    users_count = ("unique_id","count")
)
total_revenue_by_plan["average_revenue_per_customer"] = total_revenue_by_plan["total_revenue"] / total_revenue_by_plan["users_count"]
total_revenue_by_plan.sort_values(
    "average_revenue_per_customer",
    ascending = False
)

,total_revenue,users_count,average_revenue_per_customer
plan_name,,,
Business,254750.0,1019,250.0
Residential,715080.0,5959,120.0
Roam,166240.0,2078,80.0
Mini,47200.0,944,50.0


### Result

The Business plan generates the highest average monthly revenue per customer at 250, followed by Residential at 120, Roam at 80, and Mini at 50.

### Business Insight

Although Residential generates the highest total monthly revenue due to its large customer base, the Business plan generates more than twice as much average revenue per customer. This indicates that customer volume and revenue per customer contribute differently to the overall revenue performance of each plan.